packages/modular-scripts/src/__tests__/options.test.ts

```diff
@@ -0,0 +1,57 @@
+import os from 'os';
+import {
+  computeConcurrencyOption,
+  validateCompareOptions,
+} from '../utils/options';
+
+// Mocking process.exit and process.stderr.write for testing
+const exitSpy = jest.spyOn(process, 'exit').mockImplementation(() => {
+  throw new Error('Exit called');
+});
+const stderrSpy = jest.spyOn(process.stderr, 'write').mockImplementation();
+
+describe('validateCompareOptions', () => {
+  afterEach(() => {
+    exitSpy.mockClear();
+    stderrSpy.mockClear();
+  });
+
+  it('should exit process when compareBranch is defined and changed is false', () => {
+    expect(() => validateCompareOptions('branch', false)).toThrow(
+      'Exit called',
+    );
+    expect(stderrSpy).toHaveBeenCalledWith(
+      "Option --compareBranch doesn't make sense without option --changed\n",
+    );
+  });
+
+  it('should not exit process when compareBranch is undefined', () => {
+    expect(() => validateCompareOptions(undefined, false)).not.toThrow();
+    expect(exitSpy).not.toHaveBeenCalled();
+  });
+});
+
+describe('computeConcurrencyOption', () => {
+  afterEach(() => {
+    exitSpy.mockClear();
+    stderrSpy.mockClear();
+  });
+
+  it('should return the number of CPUs when concurrencyLevel is undefined', () => {
+    const cpus = os.cpus().length;
+    const result = computeConcurrencyOption(undefined);
+    expect(result).toBe(cpus || 1);
+  });
+
+  it('should exit process when concurrencyLevel is not a valid number', () => {
+    expect(() => computeConcurrencyOption('invalid')).toThrow('Exit called');
```

Comment:
We're actually missing a bug that exists in the implementation here.

It's important to test the boundary cases; in this case, there is a bug that needs fixing: the function will return `0` when `'0'` is passed when it actually should exit. This needs fixing and we need a test case to cover that. Note that the comment on line 37 of `options.ts` is also misleading and needs to be updated.